src/cqueue.rs

```diff
@@ -332,3 +332,7 @@ pub fn buffer_select(flags: u32) -> Option<u16> {
 pub fn more(flags: u32) -> bool {
     flags & sys::IORING_CQE_F_MORE != 0
 }
+
+#[cfg(feature = "unstable")]
+#[allow(non_camel_case_types)]
+pub type io_uring_buf = sys::io_uring_buf;
```

Comment:
does this seem unnecessary?

Reply:
I don't understand your question. I'm using that type in tokio-uring to use uring buffer rings.

Reply:
The layout generated by codegen can change at anytime, we need to make a stable api wrapper around it, just like https://github.com/tokio-rs/io-uring/blob/master/src/types.rs#L96

Reply:
Thanks for pointing me to an example. That will be easy to do. I'll assume adding to the bottom of the same types.rs will do. Let me know if some other ordering has been established.

Reply:
> The layout generated by codegen can change at anytime, we need to make a stable api wrapper around it, just like https://github.com/tokio-rs/io-uring/blob/master/src/types.rs#L96

Last night I said okay, I would follow the example you provided. Now I realize I should make sure I understand the reason you are asking because the layout can't be changed, its a struct in the io_uring.h file that defines the memory layout shared between the kernel and the user land. This crate is providing bindings to the syscalls just like the liburing library does and that library also has to expose struct pointers to the userland application that will work with older kernels, the current kernel, and future kernels - so the io_uring.h author can't change the layout on their end either.

I will certainly create this wrapper if it makes the use of the crate easier, but I wanted to first make sure we are both on the same page regarding what is being protected against. The io_uring.h author also knows the kernel interface can't be changed in breaking ways, forwards or backwards.

Reply:
My previous word is not correct. io_uring.h guarantees consistent layout, but codegen does not guarantee consistent api for structs.

If the type in io_uring.h changes, for example by adding a new union field, it will break api, so I say make a stable api wrapper.

Reply:
While the odds that there are a union added are low (my guess), I like your point. I'll create a stable api wrapper.

Reply:
Please check the new definition.